README.md

```diff
@@ -2,57 +2,317 @@
+
+```yaml
+visualization:
+    # Other parameters...
+    rank_qoi:
+    object_qoi:
+```
+
+While `vt-tv` natively supports a variety of QOI, such as the `load`, `id`, or `volume` of ranks and objects[^1], we also support user-defined QOI, called `attributes`.
+
+<details>
+<summary><b>Rank Attributes</b></summary>
+
+<br />
+
+Rank `attributes` are defined in the `metadata` field of the JSON data files. For example:
+
+```json
+{
+    "metadata": {
+        "rank": 0,
+        "attributes": {
+            "max_memory_usage": 8.0e+9
+        }
+    }
+}
+```
+In this example, the user defines `max_memory_usage` as a rank attribute. This can then be specified as a `rank_qoi` in the YAML configuration file.
+
+</details>
+<details>
+<summary><b>Object Attributes</b></summary>
+
+<br />
+
+Object `attributes` are defined in the `tasks` field of the JSON data files. For example:
+
+```json
+{
+    "phases": [
+        {
+            "id": 0,
+            "tasks": [
+                {
+                    "entity": {
+                        "home": 0,
+                        "id": 0,
+                        "migratable": true,
+                        "type": "object"
+                    },
+                    "node": 0,
+                    "resource": "cpu",
+                    "time": 2.0,
+                    "attributes": {
+                        "home_rank": 0,
+                        "shared_bytes": 10000.0,
+                        "shared_id": 0
+                    }
+                },
+            ]
+        }
+    ]
+}
+```
+
+In this case, the user has defined `home_rank`, `shared_bytes` and `shared_id` as potential QOI.
+
+In the YAML configuration file passed to `vt-tv`, they may specify any of these as their `object_qoi`.
+
+</details>
+
+### 2. General Structure
+
+`vt-tv` is designed according to the following hierarchy:
+
+```mermaid
+graph TD;
+    Info-->ObjectInfo;
+    Info-->Rank;
+    Rank-->PhaseWork;
+    PhaseWork-->ObjectWork;
+    ObjectWork-->ObjectCommunicator
+```
+
+Further information on each class, including methods and member variables, can be found in the documentation.
+
+#### 1. Navigating the Hierarchy
+
+Users should interact mainly with the overarching `Info` class, which contains functions that drill down the hierarchy to get the desired information.
+
+For example, an instance of `Info` holds getters to all object and rank QOI (including user_defined attributes):
+
+```cpp
+auto rank_qoi = info.getRankQOIAtPhase(rank_id, phase_id, qoi_string);
+auto obj_qoi = info.getObjectQOIAtPhase(obj_id, phase_id, qoi_string);
+```
+where the `qoi_string` is the name of the desired QOI, like "load" or "id". This string can also be a user-defined attribute, as described above.
+
+#### 2. ObjectInfo vs. ObjectWork
+
+There are two classes that hold object data: `ObjectInfo` and `ObjectWork`.
+
+`ObjectInfo` holds information about a given object across all ranks and phases. This includes:
+- the object's ID
```

Comment:
Same remark; `- the ID of the object` is more adapted to the written style.

Reply:
Fixed